src/Model.php

```diff
@@ -1247,6 +1254,31 @@ public function withID($id)
         return $this->addCondition($this->id_field, $id);
     }
 
+    /**
+     * Adds WITH cursor.
+     *
+     * @param Model  $model
+     * @param string $alias
+     * @param array  $mapping
+     * @param bool   $recursive
+     *
+     * @return $this
+     */
+    public function addWith(self $model, string $alias, array $mapping = [], bool $recursive = false)
+    {
+        if (isset($this->with[$alias])) {
+            throw new Exception(['With cursor already set with this alias', 'alias'=>$alias]);
+        }
+
+        $this->with[$alias] = [
```

Comment:
Consider using extra/data holder class with these indexes as public properties. It is more robust and it also consumes less memory.

Reply:
How can storing object be less memory consuming than just array with few elements?
Can you show small example of what you mean?

Reply:
- Robust - that's clear, class properties are refactorable
- Less memory - about a year ago we tested this and it seems that properties declared by default (i.e. not added dynamically later by assigning a non-existent one) has the names stored in the class - not instance - and the instance later is only lightweight, i.e. the property values are stored then indexed by only integer keys linked with the property names/keys stored once. It is good to keep this in mind in general, but as long as you are not storing millions of these object, the different is not huge.

Reply:
I see. Never would have thought that it can be like that.

Anyway in this case this array never will be big enough to spot the difference, so let's keep it simple. We can always refactor, but then need to be consistent in code. Otherwise it's no point to use classes in this case only.